vantage_sdk/client.py

```diff
@@ -1515,25 +1519,165 @@ def more_like_these_search(
 
     # endregion
 
-    # region Documents - Upload
+    # region Documents - Upsert Helper Functions
+
+    def _document_to_collection_compatibility_check(
+        self,
+        collection: Collection,
+        document: Union[
+            UserProvidedEmbeddingsDocument, VantageManagedEmbeddingsDocument
+        ],
+    ) -> None:
+        """
+        Checks if a document is compatible with the type of the specified collection.
+        """
+        if collection.user_provided_embeddings and isinstance(
+            document, VantageManagedEmbeddingsDocument
+        ):
+            raise ValueError(
+                f"Embeddings are required for User-provided embeddings collection. Please provide a list of {UserProvidedEmbeddingsDocument.__name__} objects."  # noqa: E501
+            )
+        elif not collection.user_provided_embeddings and isinstance(
+            document, UserProvidedEmbeddingsDocument
+        ):
+            raise ValueError(
+                f"Embeddings are not required for Vantage-managed embeddings collection. Please provide a list of {VantageManagedEmbeddingsDocument.__name__} objects."  # noqa: E501
+            )
+
+    def _upsert_documents_using_browser_upload_url(
+        self,
+        browser_upload_url: str,
+        upload_content,
+    ) -> int:
+        """
+        Uploads content to a specified collection using a browser upload URL.
+        """
+        response = requests.put(
+            browser_upload_url,
+            data=upload_content,
+        )
+
+        if response.status_code != 200:
+            raise VantageFileUploadError(response.reason, response.status_code)
+
+        return response.status_code
+
+    def _upsert_documents_from_bytes(
```

Comment:
Just a remark, we can promote this back to public if users show interest in using it.